src/server/models/Player.js

```diff
@@ -219,10 +287,34 @@ export default class Player extends Model {
     }
   }
 
+  async endOfGameCleanUp() {
+    while (this.hand.size > 0) {
+      this.moveCard(this.hand.last(), this.hand, this.deck);
+    }
+    while (this.playArea.size > 0) {
+      this.moveCard(this.hand.last(), this.playArea, this.deck);
+    }
+    while (this.discardPile.size > 0) {
+      this.moveCard(this.hand.last(), this.discardPile, this.deck);
+    }
+    while (this.tavernMat.size > 0) {
+      this.moveCard(this.hand.last(), this.tavernMat, this.deck);
+    }
+    while (this.islandMat.size > 0) {
+      this.moveCard(this.hand.last(), this.islandMat, this.deck);
+    }
+    while (this.nativeVillageMat.size > 0) {
+      this.moveCard(this.hand.last(), this.nativeVillageMat, this.deck);
+    }
+  }
+
   async play(card) {
     this.game.log(`${this.name} plays ${card.name}`);
+    this.cardsPlayedThisTurn.push({
```

Comment:
why not just
```js
this.cardsPlayedThisTurn.push(card)
```

Reply:
oh, ya, that makes so much more sense....